src/effection/events/mailbox.ts

```diff
@@ -0,0 +1,46 @@
+import { fork, monitor, Operation } from 'effection';
+import { EventEmitter } from 'events';
+
+import { on, onEach, EventName } from '../events';
+import { compile } from './pattern';
+export { any } from './pattern';
+
+export class Mailbox {
+  private subscriptions = new EventEmitter();
+  private messages = new Set();
+
+  *send(message: unknown): Operation {
+    this.messages.add(message);
+    this.subscriptions.emit('message');
+  }
+
+  *receive(pattern: unknown = undefined): Operation {
+    let match = compile(pattern);
+    let { messages, subscriptions } = this;
+    while (true) {
+      for (let message of messages) {
+        if (match(message)) {
+          messages.delete(message);
+          return message;
+        }
+      }
+      yield on(subscriptions, 'message');
```

Comment:
Still not a fan of this pattern. We've already had problems with this leaking listeners once. It's just too easy to make mistakes with this. I'd rather we attach a listener to this once.

Reply:
Another problem with this pattern, is that if we ever change this method to include any other yield points, we run the risk of missing messages. I don't like that this relies on handlers being attached/removed synchronously. Especially when the potential consequence is a deadlock.

Reply:
Refactored to manually control dispatch